__unported__/account_statement_completion_compassion/model/completion_rules.py

```diff
@@ -186,125 +221,125 @@ def get_from_lsv_dd(self, cr, uid, id, st_line, context=None):
+        # Don't gengerate invoice if it's a Sponsor gift
+        if product.categ_name == GIFT_CATEGORY:
+            res['name'] = product.name
+            contract_obj = self.env['recurring.contract']
+            contract_number = int(st_line.ref[16:21])
+            contract = contract_obj.search(
+                ['|',
+                 ('partner_id', '=', partner.id),
+                 ('correspondant_id', '=', partner.id),
+                 ('num_pol_ga', '=', contract_number),
+                 ('state', '!=', 'draft')])
+            if len(contract) == 1:
+                # Retrieve the birthday of child
+                birthdate = ""
+                if product.name == GIFT_NAMES[0]:
+                    birthdate = contract.child_id.birthdate
+                    res['child_birthdate'] = birthdate
+                    birthdate = datetime.strptime(birthdate, DF).strftime(
+                        "%d %b")
+                res['name'] += "[" + contract.child_id.code
+                res['name'] += " (" + birthdate + ")]" if birthdate else "]"
+            else:
+                res['name'] += " [Child not found] "
             return res
 
         # Setup invoice data
-        journal_ids = self.pool.get('account.journal').search(
-            cr, uid, [('type', '=', 'sale')], limit=1)
-        invoicer_id = self.pool.get('account.bank.statement').browse(
-            cr, uid, st_line['statement_id'][0], context=context
-        ).recurring_invoicer_id.id
-
+        invoicer_id = st_line.statement_id.recurring_invoicer_id.id
+        if not invoicer_id:
+            invoicer_id = self.env['recurring.invoicer'].create(
+                {'source': st_line.statement_id._name}).id
+            st_line.statement_id.write({'recurring_invoicer_id': invoicer_id})
         inv_data = {
             'account_id': partner.property_account_receivable.id,
             'type': 'out_invoice',
             'partner_id': partner.id,
-            'journal_id': journal_ids[0] if journal_ids else False,
-            'date_invoice': st_line['date'],
+            'journal_id': st_line.statement_id.journal_id.id,
+            'date_invoice': st_line.date,
             'payment_term': 1,  # Immediate payment
-            'bvr_reference': st_line['ref'],
+            'bvr_reference': st_line.ref,
             'recurring_invoicer_id': invoicer_id,
         }
 
         # Create invoice and generate invoice lines
-        invoice_obj = self.pool.get('account.invoice')
-        invoice_id = invoice_obj.create(cr, uid, inv_data, context=ctx)
-        product = self.pool.get('product.product').browse(
-            cr, uid, product_id, context=ctx)
+        invoice_obj = self.env['account.invoice']
+        invoice = invoice_obj.with_context(lang='en_US').create(inv_data)
 
         res.update(self._generate_invoice_line(
-            cr, uid, invoice_id, product, st_line, partner.id, context=ctx))
+            invoice.id, product, st_line, partner.id))
```

Comment:
The test at line 309 (`categ_name != GIFT_CATEGORY`) is useless, because this case is already excluded above.